### ISSUE TITLE
as of V1.5 docker run occasionally does not find the image - that is actually there

### ISSUE THREAD

**Issue by PeterAronZentai** [docker/docker#12642]
We started to experience this since v1.5:
```
Error response from daemon: Cannot start container cd36ac2ece76d6ddb9019abdc9f0562b54e7bc3b0e1e3c4b723fba15c1ceb838: Error getting container cd36ac2ece76d6ddb9019abdc9f0562b54e7bc3b0e1e3c4b723fba15c1ceb838 from driver devicemapper: Error mounting '/dev/mapper/docker-202:1-524289-cd36ac2ece76d6ddb9019abdc9f0562b54e7bc3b0e1e3c4b723fba15c1ceb838' on '/var/lib/docker/devicemapper/mnt/cd36ac2ece76d6ddb9019abdc9f0562b54e7bc3b0e1e3c4b723fba15c1ceb838': no such file or directory" 
```

The file is there actually when I check on it after the error is shown. so it might be a timing issue

**Comment by cressie176** [comment]
We get this consistently when deploying a container for the first time
```bash
docker ps -a
```
will show the container that failed to start. Manually restarting the container works fine. After this the problem is intermittent

**Comment by cressie176** [comment]
Same issue with 1.4.1 and 1.6. Works fine with 1.3

**Comment by cressie176** [comment]
Looks like a duplicate of https://github.com/docker/docker/issues/4036 (before clicking please notify loved ones that you will be away for sometime)

**Comment by thaJeztah** [comment]
Thanks @cressie176 to quote the important bits (and get back to your loved ones);

https://github.com/docker/docker/issues/4036#issuecomment-91689540

> You may still be affected with this issue with releases that do not include #11412 (which will be included by docker 1.7)
> There are ways to get a docker daemon th at supports udev sync. (Using a dynamically linked binary, instead of statically linked)

https://github.com/docker/docker/issues/4036#issuecomment-94034140

> If your docker daemon is using `devicemapper`, and the `docker info` shows
> "Udev sync support: false", you're gonna have a bad time. The solution is
> to never run your docker daemon with udev not in sync with devicemapper.

To test the current release-candidate (1.7.0-rc2); follow the instructions here;
https://github.com/docker/docker/pull/13528#issuecomment-109061623

The usual warnings apply; do **not** use the release candidates on critical data. In some cases, it's not possible to downgrade to 1.6.x after running docker 1.7.

Providing dynamically linked binaries is currently worked on; see this PR: https://github.com/docker/docker/pull/13680

**Comment by cpuguy83** [comment]
Closing since it looks like this was known issues with devicemapper with static binaries.
Using the new packages should be good now.

Thanks!
